src/base-apis.js

```diff
@@ -277,7 +277,21 @@ MatrixBaseApis.prototype.loginWithPassword = function(user, password, callback)
     return this.login("m.login.password", {
         user: user,
         password: password,
-    }, callback);
+    }, (error, response) => {
+        if (response && response.access_token) {
+            this._http.opts.accessToken = response.access_token;
+        }
+
+        if (response && response.user_id) {
+            this.credentials = {
+                userId: response.user_id,
+            };
+        }
+
+        if(callback) {
```

Comment:
you'll need a space after `if` to satisfy the linter

Reply:
Thanks for catching this (not sure why eslint didn't).